app/src/main/java/io/gnosis/safe/ui/transactions/details/view/TxSettingsActionView.kt

```diff
@@ -0,0 +1,113 @@
+import io.gnosis.safe.ui.transactions.getVersionForAddress
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asEthereumAddress
+
+class TxSettingsActionView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr) {
+    init {
+        orientation = LinearLayout.VERTICAL
+        gravity = Gravity.LEFT
+    }
+
+    private fun clear() {
+        removeAllViews()
+    }
+
+    fun setActionInfo(txInfo: TransactionInfo) {
+        clear()
+        val settingsChange = txInfo as TransactionInfo.SettingsChange
+        settingsMethodTitle[settingsChange.dataDecoded.method]?.let { addStringItem(it) }
+
+        val params = settingsChange.dataDecoded.parameters
+        when (settingsChange.dataDecoded.method) {
+            METHOD_CHANGE_MASTER_COPY -> {
+                val mainCopy = params.getValueByName("_masterCopy")?.asEthereumAddress()
+                val label = mainCopy?.let { it.getVersionForAddress() }
+                label?.let { addLabeledAddressItem(mainCopy, it) }
+            }
+            METHOD_CHANGE_THRESHOLD -> params.getValueByName("_threshold")?.let { addStringItem(it) }
+            METHOD_ADD_OWNER_WITH_THRESHOLD -> {
+                params.getValueByName("owner")?.asEthereumAddress()?.let { addAddressItem(it) }
+                settingsMethodTitle[METHOD_CHANGE_THRESHOLD]?.let { addStringItem(it) }
+                params.getValueByName("_threshold")?.let { addStringItem(it, R.color.dark_grey) }
+            }
+
+            METHOD_REMOVE_OWNER -> {
+                params.getValueByName("owner")?.asEthereumAddress()?.let { addAddressItem(it) }
+                settingsMethodTitle[METHOD_CHANGE_THRESHOLD]?.let { addStringItem(it) }
+                params.getValueByName("_threshold")?.let { addStringItem(it, R.color.dark_grey) }
+            }
+            METHOD_SET_FALLBACK_HANDLER -> {
+                val fallbackHandler = params.getValueByName("handler")?.asEthereumAddress()
+                val label =
+                    if (DEFAULT_FALLBACK_HANDLER == fallbackHandler) {
+                        R.string.tx_list_default_fallback_handler
+                    } else {
+                        R.string.tx_list_default_fallback_handler_unknown
+                    }
+                addLabeledAddressItem(fallbackHandler, context.getString(label))
+            }
+            METHOD_SWAP_OWNER -> {
+                params.getValueByName("oldOwner")?.asEthereumAddress()?.let { addAddressItem(it) }
+                settingsMethodTitle[METHOD_ADD_OWNER_WITH_THRESHOLD]?.let { addStringItem(it) }
+                params.getValueByName("newOwner")?.asEthereumAddress()?.let { addAddressItem(it) }
+            }
+            METHOD_ENABLE_MODULE, METHOD_DISABLE_MODULE -> {
+                params.getValueByName("module")?.asEthereumAddress()?.let { addAddressItem(it) }
+            }
+        }
```

Comment:
I am wondering if these custom views should be aware of the domain objects. This could be maybe later discussed and potentially refactored if we agree on it. I would say it is a more common pattern to pass text, assets, etc. to views and they just render it, rather than having decision trees inside them.